passes/cmds/select.cc

```diff
@@ -1394,7 +1417,12 @@ struct SelectPass : public Pass {
 				log_cmd_error("No selection to check.\n");
 			work_stack.back().optimize(design);
 			if (!work_stack.back().empty())
-				log_error("Assertion failed: selection is not empty:%s\n", sel_str.c_str());
+			{
+				RTLIL::Selection *sel = &work_stack.back();
+				sel->optimize(design);
+				std::string desc = describe_selection_for_assert(design, sel);
+				log_error("%s\nAssertion failed: selection is not empty:%s\n", desc.c_str(), sel_str.c_str());
```

Comment:
Would the following format be also be okay for you in your application?

```
log_error("Assertion failed: selection is not empty:%s\n%s", sel_str.c_str(), desc.c_str());
```

I think that output format would work better with the fact that `log_error()` appends something in front of the message, thus would modifies the first line. It is the better format imo for cases where the selection is empty. (Obviously not relevant for this error but for some of the errors below.)

Reply:
Makes no difference for me as nMigen should never generate any RTL that triggers the assertion. I've changed the format to your proposed one.